### PR TITLE
introduce multiple private states and multi-tenancy support

### PR DESCRIPTION
**Pull request by trung**
Acceptance test suites for https://github.com/ConsenSys/quorum/pull/1163

And some additional improvements:
- ability to support genesis config per node and additional tessera config per node
- ability to have mixed nodes by overriding `geth` args and genesis config

### CODE REVIEW

File: src/main/java/com/quorum/gauge/common/QuorumNetworkProperty.java [hunk 1]
```diff
@@ -296,12 +295,24 @@ public void setName(String name) {
...
+            this.psi = psi;
+        }
     }
 
     public static class DockerInfrastructureProperty {
         private boolean enabled;
         private String host;
         private Map<String, DockerContainerProperty> nodes = new HashMap<>();
+        // Quorum image provided by Terraform var.quorum_docker_image
+        private String targetQuorumImage;
```
Comment (ricardolyn) [inline src/main/java/com/quorum/gauge/common/QuorumNetworkProperty.java:313]: this is a way to modify the `develop` image by like a test image?

File: src/main/java/com/quorum/gauge/common/QuorumNode.java [hunk 2]
```diff
@@ -27,5 +27,14 @@
  */
 @Deprecated
 public enum QuorumNode {
-    Node1, Node2, Node3, Node4, Node5, Node6, Node7, Node8, Node9, Node10
+    Node1, Node1_PS1, Node1_PS2, Node1_PS3,
```
Comment (ricardolyn) [inline src/main/java/com/quorum/gauge/common/QuorumNode.java:30]: wasn't this supposed to be deprecated?

File: src/specs/01_basic/accumulator_smart_contract-bloom.spec [hunk 3]
```diff
@@ -0,0 +1,126 @@
+# Accumulator contract - exercise state/receipt/events divergence when transactions are sent to a subset of the original participants
+
+ Tags: mps-bloom
```
[the remark this answers is not in the document]
Comment (nicolae-leonte-go) [reply src/specs/01_basic/accumulator_smart_contract-bloom.spec:3]: I'm not entirely sure what we should do with this spec. It is a good way to verify that bloom filters work (even with MPS enabled) however I can't see this being part of the usual tests as it takes a very long time to run - the block height needs to reach to 4300 (around 1h for raft and a bit more for istanbul).
We can probably remove it altogether or just leave it there for people to run locally if they have any bloom concerns.
Considering the above the `basic`tag  should definitely not be applied.

File: src/main/java/com/quorum/gauge/common/QuorumNode.java (hunk 2 above)
Comment (nicolae-leonte-go) [reply src/main/java/com/quorum/gauge/common/QuorumNode.java]: I've removed the changes to this file (as the extra constants were no longer used)

File: src/specs/01_basic/accumulator_smart_contract-bloom.spec (hunk 3 above)
[the remark this answers is not in the document]
Comment (ricardolyn) [reply src/specs/01_basic/accumulator_smart_contract-bloom.spec:3]: but where is this test being run as part of the QAT CI/CD? as I don't see the tag here: https://github.com/ConsenSys/quorum-acceptance-tests/pull/89/files#diff-a3ddd7238fc6e36daeb0ef76e93fe15cc87824bd3299888075210c5ca6a474d3R117

[the remark this answers is not in the document]
Comment (ricardolyn) [reply src/specs/01_basic/accumulator_smart_contract-bloom.spec:3]: ignore, noticed now the 1h test. lets keep it like this

**Review by ricardolyn — APPROVED** [review]
👍

File: src/main/java/com/quorum/gauge/common/QuorumNetworkProperty.java (hunk 1 above)
Comment (nicolae-leonte-go) [reply src/main/java/com/quorum/gauge/common/QuorumNetworkProperty.java:313]: As far as I know it is reading the contents of the application-docker.yml generated by terraform:
```yaml
quorum:
  consensus: raft
  docker-infrastructure:
    enabled: true
    target-quorum-image: quorumengineering/quorum:develop
    target-tessera-image: quorumengineering/tessera:develop
    nodes:
      Node1:
        quorum-container-id: 14b154b5270d56653f4d5865cda816b75b8d3a1cf1b12ea82f2881bab6ce7f50
        tessera-container-id: 13a13860254e3f29532bcbfce515f034b2440a9e606b62f839fcdff26af08d44
```
When template networks are used it is possible to stop the container and update the image being used while preserving the storage (see the upgrade tests).

**Pull request merged**